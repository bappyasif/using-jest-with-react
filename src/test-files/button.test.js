import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestableButton from "../components/testable-button";

describe("App component", () => {
    // In the first test, we utilize snapshots to check whether all the nodes render as we expect them to
    it("renders magnificent monkeys", () => {
        // since screen does not have the container property, we'll destructure render to obtain container for this test
        const { container } = render(<TestableButton />);
        expect(container).toMatchSnapshot();
    });

    // In the second test, we simulate a click event. Then we check if the heading changed. toMatch is one of the various assertions we could have made.
    it("renders radical rhinos after button click", () => {
        render(<TestableButton />);
        const button = screen.getByRole("button", { name: "Click Me" });

        userEvent.click(button);

        expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
    });

    // It’s also important to note that after every test, React Testing Library unmounts the rendered components.
    // That’s why we render for each test. For a lot of tests for a component, the beforeEach jest function could prove handy.

    // Snapshot tests are fast and easy to write. One assertion saves us from writing multiple lines of code. 
    // For example, with a toMatchSnapshot, we’re spared of asserting the existence of the button and the heading.

});