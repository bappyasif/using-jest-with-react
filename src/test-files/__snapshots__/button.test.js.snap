// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <div
    id="btn-container"
  >
    TestableButton
    <button
      type="button"
    >
      Click Me
    </button>
    <h1>
      Magnificent Monkeys
    </h1>
  </div>
</div>
`;
